frontend/packages/console-app/src/components/cloud-shell/CloudShellTerminal.tsx

```diff
@@ -63,6 +66,25 @@ const CloudShellTerminal: React.FC<CloudShellTerminalProps &
 
   const { t } = useTranslation();
 
+  // wait until the web terminal is loaded.
+  // if the namespace has any problems loading then set the terminal into an unrecoverable state
+  React.useEffect(() => {
+    if (namespaceLoadError) {
+      setUnrecoverableErrorFound(true);
+      setInitError(namespaceLoadError);
+    }
+  }, [namespaceLoadError]);
```

Comment:
Do you really need a separate state for unrecoverable error? 
Isn't it just `unrecoverableErrorFound = !operatorNamespace && namespaceLoadError;` ?